### PR TITLE
Improve general performance

### PR DESCRIPTION
**Pull request by BlairArchibald**
This PR improves the general performance from around 1.7s per day to 0.6s per day.

Biggest changes includes determining if we (try to) visit a neighbour at a daily level, and various optimisations to reduce the number of iterations over the households. We also avoid extra allocations when possible.

@paulbessell Can you focus your review in particular on the probabilities for neighbour movement to make sure I've not totally messed these up!

### CODE REVIEW

File: parameters/example_population_params.json [hunk 1]
```diff
@@ -163,7 +163,7 @@
         },
         "householdProperties":{
             "pVisitorsLeaveHousehold":0.5,
-            "pHouseholdVisitsNeighbour":0.006,
+            "pHouseholdVisitsNeighbourDaily":0.144,
```
Comment (paulbessell) [inline parameters/example_population_params.json:166]: Agreed with that

File: src/main/java/uk/co/ramp/covid/simulation/place/Household.java [hunk 2]
```diff
@@ -186,23 +204,33 @@ public void doTesting(Time t) {
         }
     }
 
-    private void moveNeighbour(boolean lockdown) {
+    private void moveNeighbour(Time t, boolean lockdown) {
+        // We only visit neighbours between 9 - 20
+        if (!visitsNeighbourToday || t.getHour() < 8 || t.getHour() >= 19) {
```
Comment (paulbessell) [inline src/main/java/uk/co/ramp/covid/simulation/place/Household.java:209]: Great. we should kick these to the parameters file at some point, but looks good for now

File: src/main/java/uk/co/ramp/covid/simulation/place/Household.java [hunk 3]
```diff
@@ -186,23 +204,33 @@ public void doTesting(Time t) {
...
+        if(!lockdown || !lockCompliant) {
 	        for (Household n : getNeighbours()) {
 	            if (n.isIsolating()) {
 	                continue;
 	            }
 
-                if (PopulationParameters.get().householdProperties.pHouseholdVisitsNeighbour.sample()) {
+	            // Since we determine if we should try to visit a neighbour a the star of the day, we have
+                // equal oppertunity of visiting a neighbour each hour they are open
+                if (new Probability(1.0 / (18 - 8)).sample()) {
```
Comment (paulbessell) [inline src/main/java/uk/co/ramp/covid/simulation/place/Household.java:222]: I am not entirely sure I understand this method, but surely the neighbours are open forom 8 - 20?

**Comment by BlairArchibald** [comment]
I've not made open/close times settable which hopefully clears this up?

**Pull request merged**